artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationSyncFileMessage.java

```diff
@@ -197,6 +232,15 @@ public void decodeRest(final ActiveMQBuffer buffer) {
       }
    }
 
+   private void readFile(ByteBuffer buffer) {
+      try (FileInputStream fis = new FileInputStream(file); FileChannel channel = fis.getChannel()) {
+         channel.position(offset);
+         channel.read(buffer);
```

Comment:
For this purpose https://docs.oracle.com/javase/6/docs/api/java/nio/channels/FileChannel.html#read(java.nio.ByteBuffer, long) it more efficient. But is ok to open/close every time the FileChannel just to read a chunk of it?

Reply:
We only get here for the InVM connection where we use in our test, so it doesn't matter with the performance? If needed, we can pass another parameter into constructor like fileChannel?

Reply:
> We only get here for the InVM connection where we use in our test, so it doesn't matter with the performance

Not sure about it, EAP and other products based on Artemis use InVM connections...
maybe not with replication @jbertram @clebertsuconic any ideas?

Reply:
InVM is very important for performance. There is a use case at a company i know heavily reliant on InVM perf